.github/actions/python_prepare/action.yml

```diff
@@ -0,0 +1,48 @@
+name: "Prepare development environment"
+description: "Set up Python and Node.js environments with caching"
+runs:
+  using: "composite"
+  steps:
+    # Python setup
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.10.14
+    
+    - name: Cache Poetry installation
+      uses: actions/cache@v3
+      with:
+        path: ~/.local
+        key: poetry-${{ runner.os }}
+    
+    - name: Install Poetry
+      shell: bash
+      run: curl -sSL https://install.python-poetry.org | python3 -
+    
+    - name: Cache Poetry dependencies
+      uses: actions/cache@v3
+      with:
+        path: .venv
```

Comment:
Did you test that this cache stuff works/makes it faster? From CI it looks like  virtualenv is installed elsewhere:

```
Run poetry install
Creating virtualenv python-web3-wallet-96aaq9Ev-py3.10 in /home/runner/.cache/pypoetry/virtualenvs
```

so I wonder what is this caching.

Reply:
See CI run above, Python packages + node packages are installed only once, although the action inside `python_prepare` is called by both.